### PR TITLE
Add handlers to handle typical connection initialization requests

### PR DESCRIPTION
**Pull request by adutra**
With these new handlers, and especially with `NewDriverConnectionInitializationHandler`, I was able to make the Java driver successfully talk to `CqlServer`:

Driver:
```
try (CqlSession session =
    CqlSession.builder()
        .withLocalDatacenter("datacenter_test")
        .withKeyspace("ks1")
        .addContactPoint(InetSocketAddress.createUnresolved("127.0.0.1", 9043))
        .build()) {

  Row row = session.execute("SELECT * FROM system.local").one();

  System.out.println(row.getFormattedContents());

  session
      .getMetadata()
      .getNodes()
      .forEach(
          (uuid, node) -> {
            System.out.println(uuid);
            System.out.println(node);
          });
}
```

Server:
```
var onKeyspaceSet = func(keyspace string) {
	fmt.Println("New keyspace set: %v", keyspace)
}
server := client.NewCqlServer("127.0.0.1:9043", nil)
server.RequestHandlers = client.NewDriverConnectionInitializationHandler("cluster_test", "datacenter_test", onKeyspaceSet)
server.Start(ctx)
```

### CODE REVIEW

File: client/handlers.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				log.Debug().Msgf("%v: [set keyspace handler]: received USE %v", conn, keyspace)
+				response, _ = frame.NewResponseFrame(request.Header.Version, request.Header.StreamId, nil, nil, nil, &message.SetKeyspaceResult{Keyspace: keyspace}, false)
+			}
+		}
+		return
+	}
+}
+
+// A RequestHandler to handle USE requests. This handler intercepts REGISTER requests and replies with READY.
+var RegisterHandler RequestHandler = func(request *frame.Frame, conn *CqlServerConnection, _ RequestHandlerContext) (response *frame.Frame) {
```
Comment (joao-r-reis) [inline client/handlers.go:39]: is it necessary to assign a variable?

**Review by joao-r-reis — APPROVED** [review]

File: client/handlers.go (hunk 1 above)
Comment (adutra) [reply client/handlers.go:39]: My intent was to make it clear that this is a `RegisterHandler` and should be used as such, iow as a value – not as a function that you would call yourself.

Comment (joao-r-reis) [reply client/handlers.go:53]: Another option is to have a function that returns a `RegisterHandler` but it seems kind of pointless

Comment (adutra) [reply client/handlers.go:53]: Yeah I used that option in a few places, but only when I need to create a closure. Otherwise it's a bit pointless, I agree.

**Pull request merged**